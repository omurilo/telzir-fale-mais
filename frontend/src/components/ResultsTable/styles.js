import styled, { css } from "styled-components";

export const Container = styled.div`
  ${({ theme }) => css`
    padding: ${theme.spacing.big} ${theme.spacing.vertical};
    margin-top: ${theme.spacing.big};
    border-top: 1px solid ${theme.colors.lightGrey};
    width: 100%;
    height: auto;
    display: flex;
    justify-content: center;
    align-items: center;
    /* flex-direction: column; */
    flex-wrap: wrap;
    animation: showResults 0.8s ease-in-out;

    @media (min-width: 1280px) {
      width: 1280px;
      margin-right: auto;
      margin-left: auto;
      justify-content: space-between;
    }

    @keyframes showResults {
      from {
        width: 0;
        opacity: 0;
        height: 0;
      }
      to {
        width: 100%;
        opacity: 1;
        height: auto;
      }
    }
  `}
`;

export const Details = styled.details`
  ${({ theme }) => css`
    padding: ${theme.spacing.default};
    margin: ${theme.spacing.default};
    border: 1px solid ${theme.colors.primary};
    border-radius: ${theme.radius.default};
    width: 100%;

    & > p {
      padding: ${theme.spacing.default};

      & > strong {
        font-size: 110%;
        color: ${theme.colors.secondary};
      }
    }
  `}
`;

export const Summary = styled.summary``;

export const Info = styled.div`
  ${({ theme }) => css`
    padding: ${theme.spacing.default};
    margin: ${theme.spacing.default};
    border: 1px solid ${theme.colors.primary};
    border-radius: ${theme.radius.default};
    width: 100%;

    & > p {
      & > span {
        font-size: 110%;
        font-weight: 700;
      }

      & > strong {
        color: ${theme.colors.secondary};
      }
    }

    @media (min-width: 768px) {
      width: auto;
    }
  `}
`;
