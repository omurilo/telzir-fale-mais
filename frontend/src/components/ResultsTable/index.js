import React from "react";

import * as Styled from "./styles";

function ResultsTable({
  values: { minuteCost, faleMaisCost, withoutFaleMaisCost },
}) {
  return (
    <Styled.Container>
      <Styled.Info>
        <p>
          Custo da ligação <strong>por minuto</strong>:{" "}
          <span>{minuteCost}</span>
        </p>
      </Styled.Info>
      <Styled.Info>
        <p>
          Custo da ligação <strong>com FALE+</strong>:{" "}
          <span>{faleMaisCost}</span>
        </p>
      </Styled.Info>
      <Styled.Info>
        <p>
          Custo da ligação <strong>sem FALE+</strong>:{" "}
          <span>{withoutFaleMaisCost}</span>
        </p>
      </Styled.Info>
      <Styled.Details>
        <Styled.Summary>Taxa de minutos excedentes</Styled.Summary>
        <p>
          Os minutos excedentes tem um acréscimo de <strong>10%</strong> sobre a
          tarifa normal do minuto.
        </p>
      </Styled.Details>
    </Styled.Container>
  );
}

export default ResultsTable;
