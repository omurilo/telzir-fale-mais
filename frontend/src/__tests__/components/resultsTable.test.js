import React from "react";
import ResultsTable from "components/ResultsTable";

import { render, screen } from "../test-utils";

describe("Results Table", () => {
  it("should render the Results Table", () => {
    render(
      <ResultsTable
        values={{
          minuteCost: 0.9,
          faleMaisCost: 0,
          withoutFaleMaisCost: 35.55,
        }}
      />
    );

    expect(screen.getAllByText(/Custo da ligação/)).toMatchSnapshot();
    expect(screen.getByRole("group")).toBeInTheDocument();
    expect(screen.getByText(/Taxa de minutos excedentes/)).toBeInTheDocument();
    expect(
      screen.getByText(/Os minutos excedentes tem um acréscimo de/)
    ).toContainHTML('<p>Os minutos excedentes tem um acréscimo de <strong>10%</strong> sobre a tarifa normal do minuto.</p>');
  });
});
