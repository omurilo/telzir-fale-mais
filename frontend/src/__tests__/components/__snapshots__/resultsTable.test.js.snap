// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results Table should render the Results Table 1`] = `
Array [
  <p>
    Custo da ligação 
    <strong>
      por minuto
    </strong>
    :
     
    <span>
      0.9
    </span>
  </p>,
  <p>
    Custo da ligação 
    <strong>
      com FALE+
    </strong>
    :
     
    <span>
      0
    </span>
  </p>,
  <p>
    Custo da ligação 
    <strong>
      sem FALE+
    </strong>
    :
     
    <span>
      35.55
    </span>
  </p>,
]
`;
